src/main/java/net/openhft/chronicle/network/cluster/handlers/UberHandler.java

```diff
@@ -262,14 +262,20 @@ protected void onBytesWritten() {
     @SuppressWarnings("ForLoopReplaceableByForEach")
     protected void onWrite(@NotNull final WireOut outWire) {
         super.onWrite(outWire);
+        int removedCount = 0;
         for (int i = 0; i < writers.size(); i++) {
-            int writerIndex = (i + writerOffset) % writers.size();
+            int writerIndex = (i + writerOffset - removedCount) % writers.size();
             try {
                 if (isClosing())
                     return;
                 final WritableSubHandler<T> w = writers.get(writerIndex);
                 if (w != null)
                     w.onWrite(outWire);
+            } catch (RejectedHandlerException e) {
+                final WritableSubHandler<T> handler = writers.get(writerIndex);
+                Jvm.warn().on(getClass(), "Removing rejected handler: " + handler + ", message=" + e.getMessage(), e);
+                removeHandler(handler);
+                removedCount++;
```

Comment:
Just added RejectedHandlerException handling to onWrite because it was easy to do